hs_core/serialization.py

```diff
@@ -718,32 +716,19 @@ def write_metadata_to_resource(self, resource):
                           'phone': c.phone, 'homepage': c.homepage,
                           'researcherID': c.researcherID,
                           'researchGateID': c.researchGateID}
-                try:
-                    resource.metadata.create_element('contributor', **kwargs)
-                except IntegrityError:
-                    pass
+                resource.metadata.create_element('contributor', **kwargs)
             else:
                 msg = "Contributor with type {0} are not supported"
                 msg = msg.format(c.__class__.__name__)
                 raise TypeError(msg)
         if self.abstract:
-            try:
-                resource.metadata.create_element('description', abstract=self.abstract)
-            except IntegrityError:
-                resource.metadata.update_element('description', resource.metadata.description.id,
-                                                 abstract=self.abstract)
+            resource.metadata.create_element('description', abstract=self.abstract)
```

Comment:
@hyi Yes, this is correct.  Also, because write_metadata_to_resource() now uses django.db.transaction.atomic, it is not safe to catch database errors, as per [here](https://docs.djangoproject.com/en/1.7/topics/db/transactions/).